### DISCUSSION TITLE
First party PWA support

### DISCUSSION THREAD

**Discussion by clicktodev** [vercel/next.js#63663] [App Router]
The only decent PWA [library](https://github.com/shadowwalker/next-pwa) for nextjs is extremely outdated and doesn't support the app router which is what Vercel is pushing new projects to use. (we're forced to use Vite if PWA is necessary)
First party PWA support would be great!

**Comment by kuchta** [comment]
Good idea. BTW, I don't know if it's possilbe to implement at all, but being able to run server-side part in the SW would be also great for apps required to run entirely offline...
